### ISSUE TITLE
salt-runner-not available

### ISSUE THREAD

**Issue by swastikjohari** [saltstack/salt#63740]
I am trying to auto windows patching where enabling ,checking reboot, disabling services.
trying orchestration for it where salt-runner not available

here is the windows_update.sls:

enable_services:
  runner.state.sls:
    - mods: service_enable

check_updates:
  runner.state.sls:
    - mods: update_critical

check_reboot:
  runner.function:
    - fun: system.reboot
    - name: check_reboot

wait_for_minion:
  runner.function:
    - fun: saltutil.runner
    - name: wait_for_minion
    - args:
      - manage.up
    - kwargs:
      tgt: 'minion-patchv2-1.c.salt-training-374409.internal'

disable_services:
  runner.state.sls:
    - mods: service_disable

Logs:

[ERROR   ][22035] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.
[ERROR   ][22035] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.
[ERROR   ][22035] State 'runner.function' was not found in SLS 'windows_updates'
Reason: 'runner.function' is not available.
[ERROR   ][22035] State 'runner.function' was not found in SLS 'windows_updates'
Reason: 'runner.function' is not available.
[ERROR   ][22035] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.
[ERROR   ][23348] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.
[ERROR   ][23348] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.
[ERROR   ][23348] State 'runner.function' was not found in SLS 'windows_updates'
Reason: 'runner.function' is not available.
[ERROR   ][23348] State 'runner.function' was not found in SLS 'windows_updates'
Reason: 'runner.function' is not available.
][ERROR   ][23348] State 'runner.state' was not found in SLS 'windows_updates'

### Versions Report
Salt Version:
          Salt: 3005.1
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.0
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
        Python: 3.9.16 (main, Jan  6 2023, 22:51:58)
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: centos 7 Core
        locale: utf-8
       machine: x86_64
       release: 3.10.0-1160.83.1.el7.x86_64
        system: Linux
       version: CentOS Linux 7 Core

These 5 .sls files are stored in base: /srv/salt/
i have checked the configuration also tried adding runner_dirs: /srv/salt/ 
still no luck or else is there alternate approach for it?

**Comment by OrangeDog** [comment]
Please format code etc. with code blocks.

How did you run this orchestration? It looks like you tried `state.apply` instead.

**Comment by swastikjohari** [comment]
Hi,

enable_services:
  runner.state.sls:
    - mods: service_enable

check_updates:
  runner.state.sls:
    - mods: update_critical

check_reboot:
  runner.function:
    - fun: system.reboot
    - name: check_reboot

wait_for_minion:
  runner.function:
    - fun: saltutil.runner
    - name: wait_for_minion
    - args:
      - manage.up
    - kwargs:
        tgt: 'minion-patchv2-1.c.salt-training-374409.internal'

disable_services:
  runner.state.sls:
    - mods: service_disable

this was the actual code:

this is the error i am getting when running state.orchestrate:

[root@patch-v2 salt]# salt-run state.orchestrate windows_updates                                                                                                          
[ERROR   ] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.

[ERROR   ] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.

[ERROR   ] State 'runner.function' was not found in SLS 'windows_updates'
Reason: 'runner.function' is not available.

[ERROR   ] State 'runner.function' was not found in SLS 'windows_updates'
Reason: 'runner.function' is not available.

[ERROR   ] State 'runner.state' was not found in SLS 'windows_updates'
Reason: 'runner.state' is not available.

patch-v2.c.salt-training-374409.internal_master:
----------
          ID: enable_services
    Function: runner.state
      Result: False
     Comment: State 'runner.state' was not found in SLS 'windows_updates'
              Reason: 'runner.state' is not available.
     Changes:   
----------
          ID: check_updates
    Function: runner.state
      Result: False
     Comment: State 'runner.state' was not found in SLS 'windows_updates'
              Reason: 'runner.state' is not available.
     Changes:   
----------
          ID: check_reboot
    Function: runner.function
      Result: False
     Comment: State 'runner.function' was not found in SLS 'windows_updates'
              Reason: 'runner.function' is not available.
     Changes:   
----------
          ID: wait_for_minion
    Function: runner.function
      Result: False
     Comment: State 'runner.function' was not found in SLS 'windows_updates'
              Reason: 'runner.function' is not available.
     Changes:   
----------
          ID: disable_services
    Function: runner.state
      Result: False
     Comment: State 'runner.state' was not found in SLS 'windows_updates'
              Reason: 'runner.state' is not available.
     Changes:   

Summary for patch-v2.c.salt-training-374409.internal_master
------------
Succeeded: 0
Failed:    5
------------
Total states run:     5


 
basically, I want to update windows and check for any pending reboots wait for the minion to get online then disable the services.

I am also attaching other .sls files in /srv/salt/
[check_reboot.txt](https://github.com/saltstack/salt/files/10787690/check_reboot.txt)
[service_disable.txt](https://github.com/saltstack/salt/files/10787691/service_disable.txt)
[service_enable.txt](https://github.com/saltstack/salt/files/10787692/service_enable.txt)
[update_critical.txt](https://github.com/saltstack/salt/files/10787693/update_critical.txt)
[windows_update.txt](https://github.com/saltstack/salt/files/10787694/windows_update.txt)


Really appreciate your help

**Comment by swastikjohari** [comment]
> Hi,
> 
> enable_services: runner.state.sls: - mods: service_enable
> 
> check_updates: runner.state.sls: - mods: update_critical
> 
> check_reboot: runner.function: - fun: system.reboot - name: check_reboot
> 
> wait_for_minion: runner.function: - fun: saltutil.runner - name: wait_for_minion - args: - manage.up - kwargs: tgt: 'minion-patchv2-1.c.salt-training-374409.internal'
> 
> disable_services: runner.state.sls: - mods: service_disable
> 
> this was the actual code:
> 
> this is the error i am getting when running state.orchestrate:
> 
> [root@patch-v2 salt]# salt-run state.orchestrate windows_updates [ERROR ] State 'runner.state' was not found in SLS 'windows_updates' Reason: 'runner.state' is not available.
> 
> [ERROR ] State 'runner.state' was not found in SLS 'windows_updates' Reason: 'runner.state' is not available.
> 
> [ERROR ] State 'runner.function' was not found in SLS 'windows_updates' Reason: 'runner.function' is not available.
> 
> [ERROR ] State 'runner.function' was not found in SLS 'windows_updates' Reason: 'runner.function' is not available.
> 
> [ERROR ] State 'runner.state' was not found in SLS 'windows_updates' Reason: 'runner.state' is not available.
> 
> ## patch-v2.c.salt-training-374409.internal_master:
> ```
>       ID: enable_services
> Function: runner.state
>   Result: False
>  Comment: State 'runner.state' was not found in SLS 'windows_updates'
>           Reason: 'runner.state' is not available.
>  Changes:   
> ```
> 
> ```
>       ID: check_updates
> Function: runner.state
>   Result: False
>  Comment: State 'runner.state' was not found in SLS 'windows_updates'
>           Reason: 'runner.state' is not available.
>  Changes:   
> ```
> 
> ```
>       ID: check_reboot
> Function: runner.function
>   Result: False
>  Comment: State 'runner.function' was not found in SLS 'windows_updates'
>           Reason: 'runner.function' is not available.
>  Changes:   
> ```
> 
> ```
>       ID: wait_for_minion
> Function: runner.function
>   Result: False
>  Comment: State 'runner.function' was not found in SLS 'windows_updates'
>           Reason: 'runner.function' is not available.
>  Changes:   
> ```
> 
> ```
>       ID: disable_services
> Function: runner.state
>   Result: False
>  Comment: State 'runner.state' was not found in SLS 'windows_updates'
>           Reason: 'runner.state' is not available.
>  Changes:   
> ```
> 
> ## Summary for patch-v2.c.salt-training-374409.internal_master
> ## Succeeded: 0
> Failed: 5
> Total states run: 5
> 
> basically, I want to update windows and check for any pending reboots wait for the minion to get online then disable the services.
> 
> I am also attaching other .sls files in /srv/salt/ [check_reboot.txt](https://github.com/saltstack/salt/files/10787690/check_reboot.txt) [service_disable.txt](https://github.com/saltstack/salt/files/10787691/service_disable.txt) [service_enable.txt](https://github.com/saltstack/salt/files/10787692/service_enable.txt) [update_critical.txt](https://github.com/saltstack/salt/files/10787693/update_critical.txt) [windows_update.txt](https://github.com/saltstack/salt/files/10787694/windows_update.txt)
> 
> Really appreciate your help

code is formated only not showing here

**Comment by OrangeDog** [comment]
Observe the examples in the documentation: https://docs.saltproject.io/en/latest/topics/orchestrate/orchestrate_runner.html

The states should be `salt.state` and `salt.function`, not `runner.state,sls` nor `runner.function`.
